aten/src/ATen/native/cpu/QuickPartition.cpp

```diff
@@ -0,0 +1,568 @@
+#include <aten/src/ATen/native/cpu/QuickPartition.h>
+#include <stdint.h>
+#if CPU_CAPABILITY_AVX2
+#include <aten/src/ATen/cpu/vec256/vec256.h>
+#include <xmmintrin.h>
+#include <immintrin.h>
+#include <algorithm>
+#include <limits>
+#include <cmath>
+
+using at::vec256::Vec256;
+using at::vec256::permute;
+using at::vec256::int_same_size_t;
+
```

Comment:
Somewhere in this file we should quote the paper where this algorithm comes from.

Reply:
Ok, updated to cite that paper at the top of the file.